### PR TITLE
Move config docs to contributing guide, kill `docs` directory.

### PR DESCRIPTION
**Pull request by jchristgit**
Continuation of #521.

### CODE REVIEW

**Comment by jchristgit** [comment]
image

File: pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md [hunk 1]
```diff
@@ -143,3 +143,35 @@ Unless you are editing the Dockerfile or docker-compose.yml, you shouldn't need
...
+## Deployment
+- **`ALLOWED_HOSTS`**: A comma-separated lists of alternative hosts to allow to
+  host the website on, when `DEBUG` is not set. Optional, defaults to the
+  `pythondiscord.com` family of domains.
+
+- **`SECRET_KEY`**: The secret key used in various parts of Django. Keep this
+  secret as the name suggests! This is managed for you in debug setups.
+
+- **`STATIC_ROOT`**: The root in which `python manage.py collectstatic` collects
+  static files. Optional, defaults to `/var/www/pythondiscord.com`.
```
Comment (kosayoda) [inline pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md]: Is this still true given our current `STATIC_ROOT` default?

https://github.com/python-discord/site/blob/efa3c450f7cabaac5191c73c6f4726a1be89be23/pydis_site/settings.py#L168

**Review by kosayoda — COMMENTED** [review]
Can we have a link mentioning this new details section from the `2. Environment variables` section?

File: pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md (hunk 1 above)
Comment (jchristgit) [reply pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md]: Hello, good point, no, that's outdated, you're correct. This is from the very early pre-Docker times. I will update it.

Comment (jchristgit) [reply pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md]: See ef59812d.

**Comment by jchristgit** [comment]
I've added a link to the section from "Environment variables" and updated the static path.

**Comment by whispersofthedawn** [comment]
Might it be worth mentioning that with Docker, certain environment variables aren't necessary and handled for you? (and list those)

**Comment by jchristgit** [comment]
> Might it be worth mentioning that with Docker, certain environment variables aren't necessary and handled for you? (and list those)

Yes, I agree. However, isn't this already documented that way? The environment variable section is a subsection "Without Docker", and mentions that additional steps are needed there.

**Comment by whispersofthedawn** [comment]
> Yes, I agree. However, isn't this already documented that way? The environment variable section is a subsection "Without Docker", and mentions that additional steps are needed there.

I forgot it was under "Without Docker", my bad.

**Review by kosayoda — APPROVED** [review]

**Review by Xithrius — APPROVED** [review]

**Pull request merged**